cmd/govim/testdata/show_message.txt

```diff
@@ -0,0 +1,7 @@
+# Test that popup shows up when message is of Error type
+
+vim expr 'GOVIM_internal_ShowMessagePopup()'
+cmp stdout popup.golden
```

Comment:
We need to assert that a popup is actually shown; here we are just asserting the value returned by the function.

First, given the triggering of the `ShowMessage` is asynchronous, we need to wait until we see the call. This we do using `errlogmatch`:

https://github.com/govim/govim/blob/40a414920272565dd9dcda9de5009a492858e66d/cmd/govim/testdata/gofmt.txt#L10

You can instead wait using a pattern that will match the asynchronous `ShowMessage` callback (because you control the message that will appear in the logs).

Then you can "dump popups" to verify that the popup message shown is what you expect:

https://github.com/govim/govim/blob/40a414920272565dd9dcda9de5009a492858e66d/cmd/govim/testdata/function_hover.txt#L14

With any of this, if you want to turn on `govim` logging to stderr as you debug this test, take a look at:

https://github.com/govim/govim/wiki/govim-tests#debugging-test-failures